homeassistant/helpers/selector.py

```diff
@@ -79,6 +79,55 @@ def serialize(self) -> dict[str, dict[str, _T]]:
         return {"selector": {self.selector_type: self.config}}
 
 
+def _validate_supported_feature(supported_feature: int | str) -> int:
+    # pylint: disable=import-outside-toplevel
+    from homeassistant.components.alarm_control_panel import (
+        AlarmControlPanelEntityFeature,
+    )
+    from homeassistant.components.calendar import CalendarEntityFeature
+    from homeassistant.components.camera import CameraEntityFeature
+    from homeassistant.components.climate import ClimateEntityFeature
+    from homeassistant.components.cover import CoverEntityFeature
+    from homeassistant.components.fan import FanEntityFeature
+    from homeassistant.components.humidifier import HumidifierEntityFeature
+    from homeassistant.components.light import LightEntityFeature
+    from homeassistant.components.lock import LockEntityFeature
+    from homeassistant.components.media_player import MediaPlayerEntityFeature
+    from homeassistant.components.remote import RemoteEntityFeature
+    from homeassistant.components.siren import SirenEntityFeature
+    from homeassistant.components.update import UpdateEntityFeature
+    from homeassistant.components.vacuum import VacuumEntityFeature
+    from homeassistant.components.water_heater import WaterHeaterEntityFeature
+
+    if isinstance(supported_feature, int):
+        return supported_feature
+
+    known_entity_features = {
+        "AlarmControlPanelEntityFeature": AlarmControlPanelEntityFeature,
+        "CalendarEntityFeature": CalendarEntityFeature,
+        "CameraEntityFeature": CameraEntityFeature,
+        "ClimateEntityFeature": ClimateEntityFeature,
+        "CoverEntityFeature": CoverEntityFeature,
+        "FanEntityFeature": FanEntityFeature,
+        "HumidifierEntityFeature": HumidifierEntityFeature,
+        "LightEntityFeature": LightEntityFeature,
+        "LockEntityFeature": LockEntityFeature,
+        "MediaPlayerEntityFeature": MediaPlayerEntityFeature,
+        "RemoteEntityFeature": RemoteEntityFeature,
+        "SirenEntityFeature": SirenEntityFeature,
+        "UpdateEntityFeature": UpdateEntityFeature,
+        "VacuumEntityFeature": VacuumEntityFeature,
+        "WaterHeaterEntityFeature": WaterHeaterEntityFeature,
+    }
+
+    _, enum, feature = supported_feature.split(".", 2)
```

Comment:
you should catch if it's only has a single or no `.`

Reply:
Voluptuous actually catches this and generates an acceptable error message without catching the `ValueError` which is raised if the tuple unpacking fails:
```
voluptuous.error.MultipleInvalid: not a valid value @ data['filter'][0]['supported_features'][0]
```

Do you think it adds value to add a more specific validation error, maybe something like:
```
voluptuous.error.MultipleInvalid: Invalid supported feature 'blah', expected <domain>.<enum>.<member> @ data['filter'][0]['supported_features'][0]
```